### PR TITLE
Add Available to ResourceInfo struct

### PR DESCRIPTION
**Pull request by marquiz**
Add a new field named 'Available' to the ResourceInfo structure in order
to align with the terminology used in node status. Now ResourceInfo
is able to carry the same information as node status plus the amount of
resource currently available for new pods to be allocated (which is not
present in node status).

Also add comments describing these fields for future readers.

Based on top of #12

### CODE REVIEW

**Comment by AlexeyPerevalov** [comment]
I'm ok with this change, but need to figure out with controller-gen.

**Comment by swatisehgal** [comment]
Just thinking it would be a good idea to still expose Capacity information in the API as that would allow us to differentiate between Capacity and Allocatable meaning we would be able to learn if kube-reserved/ system-reserved have been configured on the node.

**Comment by marquiz** [comment]
> Just thinking it would be a good idea to still expose Capacity information in the API as that would allow us to differentiate between Capacity and Allocatable meaning we would be able to learn if kube-reserved/ system-reserved have been configured on the node

We can do that 👍 What do others think. @AlexeyPerevalov ?

**Comment by AlexeyPerevalov** [comment]
> 
> 
> > Just thinking it would be a good idea to still expose Capacity information in the API as that would allow us to differentiate between Capacity and Allocatable meaning we would be able to learn if kube-reserved/ system-reserved have been configured on the node
> 
> We can do that 👍 What do others think. @AlexeyPerevalov ?

I'm ok with it.

**Comment by marquiz** [comment]
OK, now updated the PR and rebased it on top of #12

File: pkg/apis/topology/v1alpha1/types.go [hunk 1]
```diff
@@ -46,25 +46,45 @@ type Zone struct {
...
 
+// ResourceInfo contains information about one resource type.
 type ResourceInfo struct {
-	Name        string             `json:"name"`
-	Allocatable intstr.IntOrString `json:"allocatable"`
-	Capacity    intstr.IntOrString `json:"capacity"`
+	// Name of the resource.
+	Name string `json:"name"`
+	// Capacity of the resource, corresponding to capacity in
+	// node status, i.e. total total amount of this resource that the node has.
```
Comment (swatisehgal) [inline pkg/apis/topology/v1alpha1/types.go]: repetition of the word `total`

File: pkg/apis/topology/v1alpha1/types.go [hunk 2]
```diff
@@ -46,25 +46,45 @@ type Zone struct {
...
+	Capacity resource.Quantity `json:"capacity"`
+	// Allocatable quantity of the resource, corresponding to allocatable in
+	// node status, i.e. total total amount of this resource available to be
```
Comment (swatisehgal) [inline pkg/apis/topology/v1alpha1/types.go]: same as above

File: pkg/apis/topology/v1alpha1/types.go (hunk 1 above)
Comment (marquiz) [reply pkg/apis/topology/v1alpha1/types.go]: Thanks, fixed!

File: pkg/apis/topology/v1alpha1/types.go (hunk 2 above)
Comment (marquiz) [reply pkg/apis/topology/v1alpha1/types.go]: Fixed

**Comment by marquiz** [comment]
Addressed review comments and rebased

**Comment by swatisehgal** [comment]
Will let @AlexeyPerevalov take a look and merge.

**Comment by marquiz** [comment]
Ping @AlexeyPerevalov

**Review by AlexeyPerevalov — APPROVED** [review]

**Pull request merged**